subprojects/core/src/integTest/groovy/org/gradle/api/tasks/MissingTaskDependenciesIntegrationTest.groovy

```diff
@@ -223,7 +217,82 @@ class MissingTaskDependenciesIntegrationTest extends AbstractIntegrationSpec {
         """
 
         expect:
-        executer.expectDocumentedDeprecationWarning(DEPRECATION_WARNING)
+        expectMissingDependencyDeprecation(":producer", ":consumer")
         succeeds("producer", "consumer")
     }
+
+    def "can have tasks which consume a filtered project directory"() {
+        file("src/main/java/MyClass.java").createFile()
+        buildFile << """
+            task firstTask {
+                def outputFile = file("build/output1.txt")
+                outputs.file(outputFile)
+                doLast {
+                    outputFile.text = "first"
+                }
+            }
+            task lastTask {
+                def outputFile = file("build/output2.txt")
+                outputs.file(outputFile)
+                doLast {
+                    outputFile.text = "last"
+                }
+            }
+            task packageProjectDir(type: Zip) {
+                from(project.projectDir) {
+                    include 'src/**'
+                }
+                destinationDirectory = file("build")
+                archiveBaseName = "output3"
+            }
+        """
```

Comment:
This test proves that the filtered inputs are actually taken into account and that we don't detect intersections in that case. I run `firstTask` and `lastTask` because the kind of check we are doing depends on whether the consumer runs first or the producer runs first.

I changed the test to only have two tasks and the test name.